### ISSUE TITLE
Ensime server starts, but stays disconnected

### ISSUE THREAD

**Issue by alex3kov** [ensime/ensime-server#714]
After `M-x ensime` it goes through usual messages of starting ensime and analyzing project, then reports that ensime is ready, but it is not connected and no ensime functionality is available in editor.
![2014-12-19-105403_1600x900_scrot](https://cloud.githubusercontent.com/assets/3616242/5501536/b0228b30-876d-11e4-8323-0bffce241b75.png)
I see no errors in *ensime-events* or *inferior-ensime-server*.
I tried reinstalling all emacs packages - same deal.

**Comment by demon386** [comment]
I encounter the same problem when I put the source code in the wrong directory (src/scala/Main.scala). Once I move it to src/main/scala/Main.scala, it connected successfully.

**Comment by aemoncannon** [comment]
That's expected. If your source is not in a valid source root, ensime won't
locate the connection.

**Comment by omnisis** [comment]
I just ran into this issue on Mac OS X.  The problem was that I had my sources in a path ~/code which was symlinked into my dropbox account folder.  I also tried using '/tmp' which on my MBPro is symlinked as well! (/private/tmp).  Only after putting it off a folder with no symlinks at all did the problem go away.  This is a pretty limiting bug in my mind since its hard to control source paths being symlinked sometimes and I got *ZERO* indication that something was wrong (ensime-events looked good, inferior-ensime looked good).  

Would it at least be possible to at least put something in the inferior window or *Messages* to indicate that one or more source paths are not the same as buffer paths for the time being until the symlink bug is fixed?
